api/src/main/java/com/microsoft/gctoolkit/GCToolKit.java

```diff
@@ -23,22 +26,24 @@
 
     private static final Logger LOGGER = Logger.getLogger(GCToolKit.class.getName());
 
-    private static JavaVirtualMachine loadJavaVirtualMachine() {
+    private JavaVirtualMachine loadJavaVirtualMachine(GCLogFile logFile) {
+        //todo: change to using ServiceLoader...
+        final String className = (logFile.isUnified()) ? "com.microsoft.gctoolkit.vertx.jvm.UnifiedJavaVirtualMachine" : "com.microsoft.gctoolkit.vertx.jvm.PreUnifiedJavaVirtualMachine";
```

Comment:
Structural issue here is probably that constructing JavaVirtualMachines should not be performed in the `api` this code means that there is a runtime dependency between the `api` and `vertx` but not enforced by maven, Fix probably should be to move JavaVirtualMachine construction to a module that has visibility of both `vertx` and `api`

Reply:
The reason it is like this was just an expedient to getting gctoolkit out in the open. Too much happens in that DefaultJavaVirtualMachine. JavaVirtualMachine impl should be decoupled from vertx. If the vertx layer needs JavaVirtualMachine, it should be injected. Or the API needed from vertx should be abstracted and injected into GCToollKit.